numpy/_core/tests/examples/cython/checks.pyx

```diff
@@ -266,3 +267,66 @@ def inc2_cfloat_struct(cnp.ndarray[cnp.cfloat_t] arr):
     # This works in both modes
     arr[1].real = arr[1].real + 1
     arr[1].imag = arr[1].imag + 1
+
+
+def npystring_pack(arr):
+    cdef char *string = "Hello world"
+    cdef size_t size = 11
+
+    allocator = cnp.NpyString_acquire_allocator(
+        <cnp.PyArray_StringDTypeObject *>cnp.PyArray_DESCR(arr)
+    )
+
+    # copy string->packed_string, the pointer to the underlying array buffer
+    if cnp.NpyString_pack(
+        allocator, <cnp.npy_packed_static_string *>cnp.PyArray_DATA(arr), string, size,
+    ) == -1:
+        return -1
```

Comment:
It'd be nice if this tested `NPyString_pack_null` too.

Reply:
You're right. Test added, and I also added a test that non-`StringDType` arrays can be passed in as well (but they get assigned `NULL` allocators).